src/TotkRstbGenerator.Core/RstbGenerator.cs

```diff
@@ -18,19 +18,27 @@ public class RstbGenerator
     private readonly string _output;
     private readonly uint _padding;
 
-    public RstbGenerator(string romfs, string? output = null, uint padding = 0)
+    public RstbGenerator(string romfs, string? sourceRstbPath = null, string? output = null, uint padding = 0)
     {
+        string path = sourceRstbPath ?? TotkConfig.Shared.RsizetablePath;
+        if (!File.Exists(path))
+        {
+            throw new FileNotFoundException(
+                sourceRstbPath is not null
+                    ? $"The file '{sourceRstbPath}' could not be found."
+                    : $"The vanilla RSTB file 'System/Resource/ResourceSizeTable.Product.{TotkConfig.Shared.Version}.rsizetable.zs'" +
+                        $" could not be found in your game dump."
+            );
+        }
+
         _romfs = romfs;
-        _output = output ?? romfs.GetRsizetableFile();
+        _output = output is not null ? 
+            Path.Combine(output, Path.GetFileName(path)) 
+            : sourceRstbPath is null ? 
+                romfs.GetRsizetableFile()
+                : Path.Combine(romfs.GetRsizetableFolder(), Path.GetFileName(path));
```

Comment:
`output` is sometimes a file path, so combining output with a name will break behaviour.

```suggestion
        _output = output is not null ? 
            output 
            : sourceRstbPath is null ? 
                romfs.GetRsizetableFile()
                : Path.Combine(romfs.GetRsizetableFolder(), Path.GetFileName(path));
```